internal/unmarshal.go

```diff
@@ -122,18 +134,127 @@ func getPointerToCopied(elemType reflect.Type, value any) reflect.Value {
 	return castedPointer
 }
 
+func GetValueFromArrowArray(arr arrow.Array, idx int) (any, error) {
+	switch castArr := arr.(type) {
+	case *array.String:
+		return castArr.Value(idx), nil
+	case *array.LargeString:
+		return castArr.Value(idx), nil
+	case *array.Uint8:
+		return castArr.Value(idx), nil
+	case *array.Uint16:
+		return castArr.Value(idx), nil
+	case *array.Uint32:
+		return castArr.Value(idx), nil
+	case *array.Uint64:
+		return castArr.Value(idx), nil
+	case *array.Int16:
+		return castArr.Value(idx), nil
+	case *array.Int32:
+		return castArr.Value(idx), nil
+	case *array.Int64:
+		return castArr.Value(idx), nil
+	case *array.Float64:
+		return castArr.Value(idx), nil
+	case *array.Boolean:
+		return castArr.Value(idx), nil
+	case *array.Date32:
+		return castArr.Value(idx).ToTime(), nil
+	case *array.Date64:
+		return castArr.Value(idx).ToTime(), nil
+	case *array.Timestamp:
+		return castArr.Value(idx).ToTime(arrow.Nanosecond), nil
+	default:
+		return nil, fmt.Errorf("unsupported array type: %T", castArr)
+	}
+}
+
+func ExtractFeaturesFromTable(table arrow.Table) ([]map[string]any, error) {
+	res := make([]map[string]any, 0)
+	reader := array.NewTableReader(table, int64(tableReaderChunkSize))
+	defer reader.Release()
+	for reader.Next() {
+		record := reader.Record()
+		for i := 0; i < int(record.NumRows()); i++ {
+			m := map[string]any{}
+			for j, col := range record.Columns() {
+				name := record.ColumnName(j)
+				if _, ok := skipUnmarshalFields[name]; ok {
+					continue
+				}
+				if _, ok := skipUnmarshalFeatureNames[getFeatureNameFromFqn(name)]; ok {
+					continue
+				}
+
+				if col.IsNull(i) {
+					m[name] = nil
+					continue
+				}
+
+				switch arr := col.(type) {
+				case *array.LargeList:
+					newSlice := make([]any, 0)
+					for ptr := arr.Offsets()[i]; ptr < arr.Offsets()[i+1]; ptr++ {
+						anyVal, valueErr := GetValueFromArrowArray(arr.ListValues(), int(ptr))
+						if valueErr != nil {
+							return nil, fmt.Errorf("error getting value for LargeList column: %w", valueErr)
+						}
+						newSlice = append(newSlice, anyVal)
+					}
+					m[name] = newSlice
+				case *array.Struct:
+					// TODO: Deserialize structs to support dataclasses
```

Comment:
Should this at least return an error right now to indicate that it's unsupported?